app/models/internal/event.rb

```diff
@@ -0,0 +1,88 @@
+module Internal
+  class Event
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+    include ApiModelConvertable
+
+    VENUE_TYPES = { add: "add", existing: "existing", none: "none" }.freeze
+
+    attribute :id, :string
+    attribute :readable_id, :string
+    attribute :status_id, :integer
+    attribute :type_id,
+              :integer,
+              default: GetIntoTeachingApiClient::Constants::EVENT_TYPES["School or University event"]
+    attribute :name, :string
+    attribute :summary, :string
+    attribute :description, :string
+    attribute :is_online, :boolean
+    attribute :start_at, :datetime
+    attribute :end_at, :datetime
+    attribute :provider_contact_email, :string
+    attribute :provider_organiser, :string
+    attribute :provider_target_audience, :string
+    attribute :provider_website_url, :string
+    attribute :building
+    attribute :venue_type, type: VENUE_TYPES, default: VENUE_TYPES[:none]
+
+    validates :name, presence: true, allow_blank: false
+    validates :readable_id, presence: true, allow_blank: false
+    validates :summary, presence: true, allow_blank: false
+    validates :description, presence: true, allow_blank: false
+    validates :summary, presence: true, allow_blank: false
+    validates :is_online, inclusion: { in: [true, false] }
+    validates :start_at, presence: true, allow_blank: false
+    validates :end_at, presence: true
+    validates :provider_contact_email,
+              presence: true,
+              allow_blank: false,
+              email_format: true,
+              length: { maximum: 100 }
+    validates :provider_organiser, presence: true, allow_blank: false
+    validates :provider_target_audience, presence: true, allow_blank: false
+    validates :provider_website_url, presence: true, allow_blank: false
+    validates_each :start_at, :end_at do |record, attr, value|
+      unless value.nil?
```

Comment:
I would maybe do this in a method instead of as a block to be consistent with the other `validate :end_after_start`